documentation/internal/Element-Type-Specs.md

```diff
@@ -152,6 +152,47 @@ _Multi-choice question, with one allowed option, displayed as Drop-down list (Co
 
 ---
 
+### Checkboxes
+
+- **type/code**: `checkbox`
+- **category**: `Question`
+
+_One or more checkboxes, any number of which can be selected/toggled_
+
+#### Input parameters
+
+- **label\***: `string` -- as above
+- **description\***: `string` -- as above [Optional]
+- **checkboxes\***: `array[string | checkbox]` -- an array of labels, one per checkbox. For more complexity, an array of Checkbox objects can be provided with the following properties:
+
+  ```
+   {
+     label: <string> - text to display next to checkbox (Can be empty string but not omitted)
+     text: <string> - value to store in Response "text" field and shown in Summary View. Will be same as label if omitted.
+     key: <string | number> - unique code used as key/property name for Response object. Defaults to numerical index of array if omitted
+     selected: <boolean> - initial state of checkbox
+   }
+  ```
+
+- **type**: `string` -- Can be "toggle" to display as a toggle switch, or "slider" to display as a slider switch (defaults to regular checkbox).
+- **layout**: `string` -- if "inline", displays checkboxes horizontally in rows. Useful if there are a lot of checkboxes.
+
+#### Response type
+
+```
+{
+  text: <string> -- comma separated list of all selected checkbox "text" values, shown in Summary view (or Review)
+  values: {
+            <key-name-1> : { text: <text value>, isSelected: <boolean>}
+            <key-name-2> : { text: <text value>, isSelected: <boolean>}
```

Comment:
Is the `key-name` the key defined in the checkbox Object as the `key`? And why this doesn't result also the label? You don't think it would be used?

Reply:
> Is the key-name the key defined in the checkbox Object as the key? 

Yes, but the _name_ of the key here is the _string_ that was given to the `key` property in the input array. See the [original post](#270) for the reasoning as to why it's like this rather than an array, along with Andrei's suggestion and discussion.

> And why this doesn't result also the label? You don't think it would be used?

If the `text` property is omitted, the label _becomes_ the text value. You only specify a text property if you want the response to store a *different* text string to what is displayed to the user in the UI.

Reply:
Ok. Thanks for the explanation. I had a quick look on issue #270 to briefly see the discussion too